### ISSUE TITLE
[MAC]: Agent upgrade failed for mac agent while upgrading from 7.17.21 BC1> 8.13.3 BC2.

### ISSUE THREAD

**Issue by amolnater-qasource** [elastic/elastic-agent#4637]
**Kibana Build details:**
```
VERSION: 8.13.3 BC2
BUILD: 72204
COMMIT: 003e4a42946390c2eb93dfb3586498ce7520a530
Artifact Link: https://staging.elastic.co/7.17.21-d107d3c3/downloads/beats/elastic-agent/elastic-agent-7.17.21-darwin-x86_64.tar.gz
```

**Host OS:**  MAC Studio

**Preconditions:**
1. 8.13.3 BC2 Kibana cloud environment should be available.
2. 7.17.21 agent should be installed.

**Steps to reproduce:**
1. Add 8.13.3 BC agent binary to the fleet settings: `https://staging.elastic.co/8.13.3-f64df5ce/downloads/`
2. Trigger upgrade for 7.17.21 agent and observe agent upgrade failed for mac agent.

**Screenshots:**
![image](https://github.com/elastic/elastic-agent/assets/77374876/09a7d02f-dd41-4342-a928-b2a2973fc3bd)

**Expected Result:**
Agent should upgrade successfully for mac agent while upgrading from 7.17.21 BC1> 8.13.3 BC2.

**Diagnostics:**
[elastic-agent-diagnostics-2024-04-30T07-12-54Z-00.zip](https://github.com/elastic/elastic-agent/files/15159746/elastic-agent-diagnostics-2024-04-30T07-12-54Z-00.zip)

**Comment by amolnater-qasource** [comment]
@manishgupta-qasource Please review.

**Comment by manishgupta-qasource** [comment]
Secondary review for this ticket is Done

**Comment by pierrehilbert** [comment]
Hey @amolnater-qasource,
I didn't check in details for know but could it be related to https://github.com/elastic/elastic-agent/issues/4613?

**Comment by cmacknz** [comment]
Yes I see defend in the policy, almost certainly this is related to https://github.com/elastic/elastic-agent/issues/4613.

If you remove defend from the policy does it succeed? If so its a known issue.

**Comment by amolnater-qasource** [comment]
Hi @cmacknz 

Thank you for the update.

Yes, we have revalidated this by upgrading the agent from 7.17.21 BC1>8.13.3 BC2 without Elastic Defend and the agent upgraded successfully.

**Screenshots:**
![Screenshot 2024-05-01 at 10 19 35 AM](https://github.com/elastic/elastic-agent/assets/77374876/786fa102-d06e-4c25-9a8e-e3caeddd7657)
![Screenshot 2024-05-01 at 10 27 08 AM](https://github.com/elastic/elastic-agent/assets/77374876/33541e79-e033-432e-9113-9f3d865610f6)



We can close this issue if the reported issue already exists.

Please let us know if anything else is required from our end.
Thanks!!

**Comment by cmacknz** [comment]
Thanks for confirming! We'll use https://github.com/elastic/elastic-agent/issues/4613 to track this, closing.
